src/enhancers/withState.js

```diff
@@ -1,18 +1,26 @@
-import {useState} from 'react';
+import {useState, useMemo} from 'react';
+import {PROPS} from '../util/constants';
 
-export default (stateName, stateUpdaterName, initialValue) => ({
+export const withState = (stateName, stateUpdaterName, initialValue) => ({
   generateNewVariable,
 }) => {
   const initialValueAlias = generateNewVariable();
+  const initialValueIsFunctionAlias = generateNewVariable();
+  const computedInitialValueAlias = generateNewVariable();
   const stateValue = generateNewVariable();
 
   return {
     dependencies: {
       useState,
+      useMemo,
       [initialValueAlias]: initialValue,
+      [initialValueIsFunctionAlias]: Object.prototype.toString.call(initialValue) === '[object Function]',
     },
     initialize: `
-      const ${stateValue} = useState(${initialValueAlias});
+      const ${computedInitialValueAlias} = ${initialValueIsFunctionAlias}
```

Comment:
In thinking about the implementation for this, I'm leaning towards doing the check in the higher code and emitting either code set appropriately

Reply:
Yeah that works too, I went back and forth with some different implementations. I'll update this a bit later